### PR TITLE
Merge psa api branch into development

### PR DESCRIPTION
**Pull request by yanesca**
There were no merge conflicts. (Thanks to the recent merge of development into the API branch.)

This PR has to be merged in tandem with https://github.com/ARMmbed/mbedtls/pull/2767.

### CODE REVIEW

**Comment by yanesca** [comment]
The crypto CI only fails on the API check, which is correct, because we are changing the API and the TLS tests are not expected to pass without https://github.com/ARMmbed/mbedtls/pull/2767.

File: library/psa_crypto.c [hunk 1]
```diff
@@ -893,124 +988,228 @@ static psa_status_t psa_remove_key_data_from_memory( psa_key_slot_t *slot )
...
-        return( PSA_ERROR_TAMPERING_DETECTED );
+        return( PSA_ERROR_CORRUPTION_DETECTED );
     }
 
     return( PSA_SUCCESS );
 }
 
+static void psa_abort_operations_using_key( psa_key_slot_t *slot )
+{
+    /*FIXME how to implement this?*/
```
[the remark this answers is not in the document]
Comment (yanesca) [reply library/psa_crypto.c]: I think we are happy with some code path not being fully complete in Mbed Crypto and PSA support is still marked as experimental in Mbed TLS.

Regarding the comment itself, @gilles-peskine-arm prefers to keep them and I can't see any benefit of removing them either.

Still being consistent and call all of the comments of this kind `TODO` every time would look better. I'll add a commit with fixing this and all the `TOnogrepDO`s.

[the remark this answers is not in the document]
Comment (gilles-peskine-arm) [reply library/psa_crypto.c]: Let's merge this into development and follow up with https://github.com/ARMmbed/mbed-crypto/pull/221 soon after.

[the remark this answers is not in the document]
Comment (k-stachowiak) [reply library/psa_crypto.c]: Thanks for addressing this. However, I have seen a todo issue checker in one version of the crypto library. If that is placed in the development branch, the todos will get pointed out as issues. Is that planned to happen, or has it been removed from `check-files.py`?

I don't mind comments like this, I just wanted to raise this as a potential unwanted change; maybe some task was not finished, or the necessary changes have got lost by the merge algorithm.

If this is supposed not to be implemented yet, I don't mind the way it is commented, especially that "FIXME" is one of the canonical terms.

[the remark this answers is not in the document]
Comment (yanesca) [reply library/psa_crypto.c]: I can't find the word `TODO` in `check-files.py` and @gilles-peskine-arm said that he remembers that we removed that check. So I think it shouldn't be a problem.

[the remark this answers is not in the document]
Comment (gilles-peskine-arm) [reply library/psa_crypto.c]: `check-files.py` used to proscribe “TODO” but we removed this (#2587) and I used “TOnogrepDO” while waiting for that change to propagate to the API branch.

**Comment by yanesca** [comment]
@k-stachowiak , @AndrzejKurek Thank you very much for the reviews.

@gilles-peskine-arm Can we merge this?

**Comment by gilles-peskine-arm** [comment]
@yanesca I haven't looked at the code, but it's a straightforward merge and CI is happy. I don't want to merge this until https://github.com/ARMmbed/mbedtls/pull/2767 is ready though.

I'd rather not merge https://github.com/ARMmbed/mbedtls/pull/2767 until we fully understand the implications on our users. This is an API change, even if it's in an API (`psa_xxx`) that few of our users use yet. What's the impact on Mbed OS? On TF-M? We should make sure we understand the impact or are prepared to deal with anything that breaks. Since I won't be around for the next two weeks, I'd rather hand that hot potato to @Patater.

**Comment by Patater** [comment]
Tested merge at https://jenkins-internal.mbed.com/blue/organizations/jenkins/mbedtls-psa-release-new/detail/mbedtls-psa-release-new/497/

Mbed OS testing didn't work due to the CI system having a bug, but otherwise all tests passing.

**Pull request merged**